x-pack/plugins/apm/public/components/shared/environment_filter/index.tsx

```diff
@@ -5,106 +5,60 @@
+import { EnvironmentSelect } from '../environment_select';
+import { ServiceEnvironmentSelect } from '../service_environment_select';
 
 function updateEnvironmentUrl(
   history: History,
   location: ReturnType<typeof useLocation>,
-  environment: string
+  environment?: string
 ) {
   const nextEnvironmentQueryParam = environment;
   history.push({
     ...location,
     search: fromQuery({
       ...toQuery(location.search),
-      environment: nextEnvironmentQueryParam,
+      ...(environment &&
+        environment.length > 0 && { environment: nextEnvironmentQueryParam }),
     }),
   });
 }
 
-const SEPARATOR_OPTION = {
-  text: `- ${i18n.translate(
-    'xpack.apm.filter.environment.selectEnvironmentLabel',
-    { defaultMessage: 'Select environment' }
-  )} -`,
-  disabled: true,
-};
-
-function getOptions(environments: string[]) {
-  const environmentOptions = environments
-    .filter((env) => env !== ENVIRONMENT_NOT_DEFINED.value)
-    .map((environment) => ({
-      value: environment,
-      text: environment,
-    }));
-
-  return [
-    ENVIRONMENT_ALL_SELECT_OPTION,
-    ...(environments.includes(ENVIRONMENT_NOT_DEFINED.value)
-      ? [ENVIRONMENT_NOT_DEFINED]
-      : []),
-    ...(environmentOptions.length > 0 ? [SEPARATOR_OPTION] : []),
-    ...environmentOptions,
-  ];
-}
-
 export function ApmEnvironmentFilter() {
-  const { status, environments, environment } = useEnvironmentsContext();
-
-  return (
-    <EnvironmentFilter
-      status={status}
-      environment={environment}
-      environments={environments}
-    />
-  );
-}
-
-export function EnvironmentFilter({
-  environment,
-  environments,
-  status,
-}: {
-  environment: Environment;
-  environments: Environment[];
-  status: FETCH_STATUS;
-}) {
+  const { environment, serviceName, start, end } = useEnvironmentsContext();
   const history = useHistory();
   const location = useLocation();
 
-  // Set the min-width so we don't see as much collapsing of the select during
-  // the loading state. 200px is what is looks like if "production" is
-  // the contents.
-  const minWidth = 200;
-
-  const options = getOptions(environments);
+  const prepend = i18n.translate('xpack.apm.filter.environment.label', {
+    defaultMessage: 'Environment',
+  });
 
-  return (
-    <EuiSelect
-      fullWidth
-      prepend={i18n.translate('xpack.apm.filter.environment.label', {
-        defaultMessage: 'Environment',
-      })}
-      options={options}
-      value={environment}
-      onChange={(event) => {
-        updateEnvironmentUrl(history, location, event.target.value);
-      }}
-      isLoading={status === FETCH_STATUS.LOADING}
-      style={{ minWidth }}
-      data-test-subj="environmentFilter"
+  return serviceName ? (
+    <ServiceEnvironmentSelect
+      prepend={prepend}
+      onChange={(changeValue) =>
+        updateEnvironmentUrl(history, location, changeValue)
```

Comment:
This updates the url when the input is submitted (on enter). But we should only update the url when a specific option is selected.